edu-services/gradebook-service/impl/src/java/org/sakaiproject/component/gradebook/GradebookServiceHibernateImpl.java

```diff
@@ -201,6 +200,16 @@ public Object doInHibernate(Session session) throws HibernateException {
 			log.warn("AUTHORIZATION FAILURE: User " + getUserUid() + " in gradebook " + gradebookUid + " attempted to get assignment " + assignmentName);
 			throw new SecurityException("You do not have permission to perform this operation");
 		}
+		if (NumberUtils.isNumber(assignmentName)) {
+			try {
+				return getAssignment(gradebookUid, NumberUtils.toLong(assignmentName, -1L));
```

Comment:
Chuck a log in here to assist debugging and indicate that it is using the conversion method?